pkg/validate/validate.go

```diff
@@ -29,6 +29,16 @@ var validAgentPoolProfileNames = map[string]struct{}{
 	string(api.AgentPoolProfileRoleMaster):  struct{}{},
 }
 
+// az account list-locations will fetch the latest additions, if needed
+var azureZones = regexp.MustCompile(`{` +
+	`eastasia|southeastasia|centralus|eastus|eastus2|westus|northcentralus|` +
+	`southcentralus|northeurope|westeurope|japanwest|japaneast|brazilsouth|` +
+	`australiaeast|australiasoutheast|southindia|centralindia|westindia|` +
```

Comment:
I don't think we want to hard-code these ourselves and I'm not aware of a canonical location to pull these from.  Think it'll have to be something like `\w+\.cloudapp\.azure\.com`

Reply:
looks like there's an api to retrieve these from
`
az account list-locations | jq .[].name --raw-output
`

Reply:
@charlesakalugwu true but how can we fetch it dynamically into the code without
1) requiring always on-line
2) requiring az credentials

I guess I will follow Jim's suggestion, which was my initial solution